packages/compiler/src/server/diagnostics.ts

```diff
@@ -175,3 +201,75 @@ function convertSeverity(severity: "warning" | "error"): DiagnosticSeverity {
       return DiagnosticSeverity.Error;
   }
 }
+
+async function getDiagnosticInTspConfig(
+  fileService: FileService,
+  configFilePath: string | undefined,
+  readFile: ((path: string) => Promise<SourceFile>) | undefined,
+  emitterName: string | undefined,
+): Promise<VSLocation | typeof NoTarget | undefined> {
+  if (configFilePath && readFile && emitterName && emitterName.length > 0) {
+    const docTspConfig = fileService.getOpenDocument(configFilePath);
+    if (!docTspConfig) {
+      return NoTarget;
+    }
+
+    const range = await getDiagnosticRangeInTspConfig(configFilePath, readFile, emitterName);
+    if (range === undefined) {
+      return NoTarget;
+    }
+    return {
+      range,
+      document: docTspConfig,
+    };
+  }
+  return undefined;
+}
+
+export async function getDiagnosticRangeInTspConfig(
+  configFilePath: string,
+  readFile: (path: string) => Promise<SourceFile>,
+  emitterName: string,
+): Promise<Range | undefined> {
+  const [yamlScript] = parseYaml(await readFile(configFilePath));
+  const target = getLocationInYamlScript(yamlScript, ["emit", emitterName], "key");
+  if (target.pos === 0) {
+    return undefined;
+  }
+
+  const lineAndChar = target.file.getLineAndCharacterOfPosition(target.pos);
+  return VSRange.create(
+    lineAndChar.line,
+    lineAndChar.character,
+    lineAndChar.line,
+    lineAndChar.character + emitterName.length,
+  );
+}
+
+function getDiagnosticInVsCodeSettings(
+  diagnostic: Diagnostic,
+  document: TextDocument,
+  clientConfig?: Config | undefined,
+  emitterName?: string | undefined,
+): [VSDiagnostic, TextDocument][] {
+  let customMsg = "";
+  if (clientConfig?.lsp?.emit && clientConfig.lsp.emit.includes(emitterName || "")) {
+    customMsg = " [It's configured in vscode settings]";
+  } else {
+    customMsg = " [No target source location reported for this Error/Warning]";
```

Comment:
this should not be handled here. the diagnostic can't be recognized in settings doesn't mean it can't be found in other target source location.